### PR TITLE
[CELEBORN-973] Improve HttpRequestHandler handle HTTP request with base, master and worker

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

The code that `HttpRequestHandler` handles HTTP request could be improved with handling HTTP request with base, master and worker.

### Why are the changes needed?

Improves `HttpRequestHandler` handle HTTP request with base, master and worker.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Internal tests.

### CODE REVIEW

File: service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala [hunk 1]
```diff
@@ -43,11 +47,11 @@ class HttpRequestHandler(
     val uri = req.uri()
     val msg = handleRequest(uri)
     val response = msg match {
-      case INVALID =>
+      case "invalid" =>
         if (prometheusHttpRequestHandler != null) {
           prometheusHttpRequestHandler.handleRequest(uri)
         } else {
-          s"invalid uri ${uri}"
+          s"invalid uri $uri"
```
Comment (AngersZhuuuu) [inline service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala:54]: Could we add a help info here? We can do this in a separated pr after this pr.  I have raise a ticket https://issues.apache.org/jira/browse/CELEBORN-829

**Review by AngersZhuuuu — APPROVED** [review]
Nice change! LGTM

File: service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala (hunk 1 above)
Comment (SteNicholas) [reply service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala:54]: @AngersZhuuuu, I will raise another pull request to follow above help info.

**Review by pan3793 — APPROVED** [review]

**Review by RexXiong — APPROVED** [review]
Thanks @SteNicholas  , besides this pr, One more thing is that we can go more deeper for HttpService, currently the two implements of HttpService have many different parts, Celeborn can keep common part and separate those different part to Master/Worker HttpService，then In MasterHttpRequestHandler can directly use `Master` Service,  and Master Service needn't have to know/care those none related methods, the same thing we can do in WorkerHttpRequestHandler too, to implements this, however, need move the corresponding handler into related module,  may be this can be done in future. For this pr it's ok to merge.

**Comment by pan3793** [comment]
Thanks @SteNicholas and all, merged to main for 0.4

File: service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala [hunk 2]
```diff
@@ -27,14 +27,18 @@ import org.apache.celeborn.common.internal.Logging
 import org.apache.celeborn.common.metrics.sink.PrometheusHttpRequestHandler
 import org.apache.celeborn.server.common.{HttpService, Service}
 
+/**
+ * A handler for the REST API that defines how to handle the HTTP request given a message.
+ *
+ * @param service The service of HTTP server.
+ * @param uri The uri of HTTP request.
```
Comment (cxzl25) [inline service/src/main/scala/org/apache/celeborn/server/common/http/HttpRequestHandler.scala:34]: nit: wrong param description in doc

**Pull request closed without merging**